src/main/java/net/fabricmc/loader/metadata/ModMetadataParser.java

```diff
@@ -51,17 +51,22 @@
 	private static final JsonParser JSON_PARSER = new JsonParser();
 
 	private static LoaderModMetadata getMod(FabricLoader loader, JsonObject object) {
-		if (!object.has("schemaVersion")) {
-			return GSON_V0.fromJson(object, ModMetadataV0.class);
-		} else {
-			//noinspection SwitchStatementWithTooFewBranches
-			switch (object.get("schemaVersion").getAsInt()) {
-				case 1:
-					return GSON_V1.fromJson(object, ModMetadataV1.class);
-				default:
-					loader.getLogger().warn("Mod ID " + (object.has("id") ? object.get("id").getAsString() : "<unknown>") + " has invalid schema version: " + object.get("schemaVersion").getAsInt());
-					return null;
+		try {
+			if (!object.has("schemaVersion")) {
+				return GSON_V0.fromJson(object, ModMetadataV0.class);
+			} else {
+				//noinspection SwitchStatementWithTooFewBranches
+				switch (object.get("schemaVersion").getAsInt()) {
+					case 1:
+						return GSON_V1.fromJson(object, ModMetadataV1.class);
+					default:
+						loader.getLogger().warn("Mod ID " + (object.has("id") ? object.get("id").getAsString() : "<unknown>") + " has invalid schema version: " + object.get("schemaVersion").getAsInt());
+						return null;
+				}
 			}
+		} catch (Throwable t) {
+			// Catch all exceptions that could occur from parsing metadata and rethrow as JsonSyntaxException.
+			throw new JsonParseException(t);
```

Comment:
Just FYI you probably don't need to make any changes to this file as it's all caught in `ModResolver`.